src/mame/includes/tsconf.h

```diff
@@ -0,0 +1,212 @@
+// license:BSD-3-Clause
+// copyright-holders:Andrei I. Holub
+/*****************************************************************************
+ *
+ * includes/tsconfig.h
+ *
+ ****************************************************************************/
+#ifndef MAME_INCLUDES_TSCONF_H
+#define MAME_INCLUDES_TSCONF_H
+
+#pragma once
+
+#include "emu.h"
+#include "machine/beta.h"
+#include "machine/spi_sdcard.h"
+#include "spectrum.h"
+#include "spec128.h"
+#include "machine/tsconfdma.h"
+#include "machine/nvram.h"
+#include "machine/pckeybrd.h"
+#include "machine/8042kbdc.h"
+#include "tilemap.h"
+
+#define LOG_ERROR (1U << 0)
+#define LOG_WARN (1U << 1)
+#define LOG_DEBUG (1U << 2)
+
+#define VERBOSE (LOG_WARN)
+#define LOG_OUTPUT_FUNC osd_printf_info
+
+#include "logmacro.h"
+#define LOGERROR(...) LOGMASKED(LOG_DEBUG | LOG_WARN | LOG_ERROR, __VA_ARGS__)
+#define LOGWARN(...) LOGMASKED(LOG_DEBUG | LOG_WARN, __VA_ARGS__)
+#define LOGDEBUG(...) LOGMASKED(LOG_DEBUG, __VA_ARGS__)
+
+#define PAGE(_r) ((_r) << 14)
+#define RAM_PAGE_OFFST(_page, _offset) (PAGE(_page) + _offset)
+
+#define REG(_p) m_regs[_p]
+#define REGNUM(_r) (&(_r) - &(V_CONFIG))
+
+#define V_CONFIG REG(0x00)
+#define V_PAGE REG(0x01)
+
+#define G_X_OFFS_L REG(0x02)
+#define G_X_OFFS_H REG(0x03)
+#define G_Y_OFFS_L REG(0x04)
+#define G_Y_OFFS_H REG(0x05)
+
+#define TS_CONFIG REG(0x06)
+#define PAL_SEL REG(0x07)
+#define BORDER REG(0x0f)
+
+#define PAGE0 REG(0x10)
+#define PAGE1 REG(0x11)
+#define PAGE2 REG(0x12)
+#define PAGE3 REG(0x13)
+#define FMAPS REG(0x15)
+
+#define T_MAP_PAGE REG(0x16)
+#define T0_G_PAGE REG(0x17)
+#define T1_G_PAGE REG(0x18)
+#define SG_PAGE REG(0x19)
+
+#define DMAS_ADDRESS_L REG(0x1a)
+#define DMAS_ADDRESS_H REG(0x1b)
+#define DMAS_ADDRESS_X REG(0x1c)
+#define DMAD_ADDRESS_L REG(0x1d)
+#define DMAD_ADDRESS_H REG(0x1e)
+#define DMAD_ADDRESS_X REG(0x1f)
+#define DMA_WAIT_PORT_DEV REG(0x25)
+#define DMA_LEN REG(0x26)
+#define DMA_CTRL REG(0x27)
+#define DMA_NUM_L REG(0x28)
+#define DMA_NUM_H REG(0x2c)
+
+#define SYS_CONFIG REG(0x20)
+#define MEM_CONFIG REG(0x21)
+#define HS_INT REG(0x22)
+#define VS_INT_L REG(0x23)
+#define VS_INT_H REG(0x24)
+
+#define INT_MASK REG(0x2a)
+
+#define T0_X_OFFSER_L REG(0x40)
+#define T0_X_OFFSER_H REG(0x41)
+#define T0_Y_OFFSER_L REG(0x42)
+#define T0_Y_OFFSER_H REG(0x43)
+#define T1_X_OFFSER_L REG(0x44)
+#define T1_X_OFFSER_H REG(0x45)
+#define T1_Y_OFFSER_L REG(0x46)
+#define T1_Y_OFFSER_H REG(0x47)
```

Comment:
Please don’t leak all this stuff in the global namespace.  Use helper member functions so it’s scoped properly and produces better debugger symbols.  Also, don’t use `SCREAMING_SNAKE_CASE` for things that aren’t actually constants or function-like macros – it will make it more consistent with the rest of the code.

Reply:
Is it ok to use SCREAMING_SNAKE_CASE in enums?

Reply:
Still thinking how to represents registers map
* enum
* or struct

Any preferences?

Reply:
It’s OK to use `SCREAMING_SNAKE_CASE` for enum constants – that’s what we usually do.

As for how to represent it, it depends:
* If you regularly need to access registers by index, then store them in an array, and either use constants for register indices or helper member functions to access registers with meaningful names.
* If you don’t need to access registers by index, a struct may be simpler, and it lets you add additional behaviour encapsulated with the storage more easily.